src/In.ProjectEKA.HipService/Link/Patient/LinkPatient.cs

```diff
@@ -135,5 +93,34 @@ into careContextPatient
             }).ValueOr(new Tuple<PatientLinkResponse, ErrorResponse>(null,
                 new ErrorResponse(new Error(ErrorCode.CareContextNotFound, "Care Context Not Found"))));
         }
+
+        private Tuple<Patient, ErrorResponse> PatientAndCareContextValidation(
+            PatientLinkReferenceRequest request)
+        {
+            return patientRepository.PatientWith(request.Patient.ReferenceNumber).Map(
+                patient =>
+                {
+                    var programs = (from careContext in request.Patient.CareContexts
+                        where patientRepository
+                            .ProgramInfoWith(request.Patient.ReferenceNumber, careContext.ReferenceNumber)
+                            .HasValue
+                        select patientRepository.ProgramInfoWith(request.Patient.ReferenceNumber,
+                            careContext.ReferenceNumber)
+                        into careContextPatient
+                        select careContextPatient.Map(context => context)).ToList();
+                    if (programs.Count != request.Patient.CareContexts.Count())
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Add curly braces around the nested statement(s) in this 'if' block.](https://app.codacy.com/gh/ProjectEKA/hip-service/pullRequest?prid=4839081)